stdlib/universe/triple_exponential_derivative_test.go

```diff
@@ -0,0 +1,680 @@
+						{execute.Time(7), uint64(7)},
+						{execute.Time(8), uint64(8)},
+						{execute.Time(9), uint64(9)},
+						{execute.Time(10), uint64(10)},
+						{execute.Time(11), uint64(11)},
+						{execute.Time(12), uint64(12)},
+						{execute.Time(13), uint64(13)},
+						{execute.Time(14), uint64(14)},
+						{execute.Time(15), uint64(15)},
+						{execute.Time(16), uint64(14)},
+						{execute.Time(17), uint64(13)},
+						{execute.Time(18), uint64(12)},
+						{execute.Time(19), uint64(11)},
+						{execute.Time(20), uint64(10)},
+						{execute.Time(21), uint64(9)},
+						{execute.Time(22), uint64(8)},
+						{execute.Time(23), uint64(7)},
+						{execute.Time(24), uint64(6)},
+						{execute.Time(25), uint64(5)},
+						{execute.Time(26), uint64(4)},
+						{execute.Time(27), uint64(3)},
+						{execute.Time(28), uint64(2)},
+						{execute.Time(29), uint64(1)},
+					},
+				},
+			}},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(11), float64(18.181818181818187)},
+					{execute.Time(12), float64(15.384615384615374)},
+					{execute.Time(13), float64(13.33333333333333)},
+					{execute.Time(14), float64(11.764705882352944)},
+					{execute.Time(15), float64(10.526315789473696)},
+					{execute.Time(16), float64(8.304761904761904)},
+					{execute.Time(17), float64(5.641927541329594)},
+					{execute.Time(18), float64(3.0392222148231784)},
+					{execute.Time(19), float64(0.716067574030288)},
+					{execute.Time(20), float64(-1.2848911076603242)},
+					{execute.Time(21), float64(-2.9999661985600445)},
+					{execute.Time(22), float64(-4.493448741755913)},
+					{execute.Time(23), float64(-5.836238000516913)},
+					{execute.Time(24), float64(-7.099092024379772)},
+					{execute.Time(25), float64(-8.352897627933453)},
+					{execute.Time(26), float64(-9.673028502435233)},
+					{execute.Time(27), float64(-11.147601363985949)},
+					{execute.Time(28), float64(-12.891818138458877)},
+					{execute.Time(29), float64(-15.074463280730022)},
+				},
+			}},
+		},
+		{
+			name: "not enough values for first ema",
+			spec: &universe.TripleExponentialDerivativeProcedureSpec{
+				N: 10,
+			},
+			data: []flux.Table{&executetest.Table{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), float64(1)},
+					{execute.Time(2), float64(2)},
+					{execute.Time(3), float64(3)},
+					{execute.Time(4), float64(4)},
+					{execute.Time(5), float64(5)},
+					{execute.Time(6), float64(6)},
+					{execute.Time(7), float64(7)},
+				},
+			},
+			},
+			want: []*executetest.Table{{
+				ColMeta: []flux.ColMeta{
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(7), math.NaN()},
```

Comment:
This is something that I also discussed with Nathaniel. I believe his reasoning was that an incomplete period has no meaning. Not sure how it is represented in flux, but in the tables it appears as a blank

Reply:
No meaning and unknown are usually represented by nulls 🤔 
@nathanielc can you comment on this?

Reply:
I think  I agree here. An incomplete period has no meaning  so a null value would be more meaningful to the user since it represents missing or unknown values.